### PR TITLE
Fix get_tags() method

### PR DESCRIPTION
**Pull request by gabegma**
## Description:
* Major issue that I introduced a while ago: `get_tags()` could take minutes now to compute on bigger datasets. This will fix a major bug where the app would no longer load on bigger datasets.
* You can see the profiling before and after.
[logs_before.txt](https://github.com/ServiceNow/azimuth/files/11215771/logs_before.txt)
[logs.txt](https://github.com/ServiceNow/azimuth/files/11215772/logs.txt)

## Checklist

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

File: azimuth/dataset_split_manager.py [hunk 1]
```diff
@@ -326,10 +326,7 @@ def get_tags(
             ds = ds.select(indices)
         available_tags = self._tags if table_key is None else self._tags + self._prediction_tags
 
-        return {
-            row_idx: {tag: ds[tag][idx] for tag in available_tags}
-            for idx, row_idx in enumerate(ds["row_idx"])
-        }
+        return {row[DatasetColumn.row_idx]: {tag: row[tag] for tag in available_tags} for row in ds}
```
Comment (JosephMarinier) [inline azimuth/dataset_split_manager.py:329]: Do you think that `ds.map()` could be faster than a `for` loop on `ds`? Something like this (not tested):
```suggestion
        return dict(
            ds.map(
                lambda row: (row[DatasetColumn.row_idx], {tag: row[tag] for tag in available_tags}),
                input_columns=[DatasetColumn.row_idx, *available_tags],
            )
        )
```

Comment (gabegma) [reply azimuth/dataset_split_manager.py:329]: I'll profile and get back to you!!

Comment (JosephMarinier) [reply azimuth/dataset_split_manager.py:329]: In fact, I just realized that in the two places that we call `get_tags()`, we then do `.values()`, so I think we can avoid the `dict` altogether!
```suggestion
        ds.map(
            lambda row: {tag: row[tag] for tag in available_tags},
            input_columns=[DatasetColumn.row_idx, *available_tags],
        )
```

Comment (gabegma) [reply azimuth/dataset_split_manager.py:329]: The goal was to be consistent with the `add_tags()` method - might not be needed though.

Comment (JosephMarinier) [reply azimuth/dataset_split_manager.py:329]: I see.

Comment (gabegma) [reply azimuth/dataset_split_manager.py:329]: I get `TypeError: <lambda>() takes 1 positional argument but 24 were given`. I'm not familiar enough with the `map() ` function to debug quickly.

Comment (JosephMarinier) [reply azimuth/dataset_split_manager.py:329]: Oh, right! Forget my `input_columns`. Without it, it should work, I think.

Comment (JosephMarinier) [reply azimuth/dataset_split_manager.py:329]: I am not familiar at all. I might be completely off. We can forget about it.

**Review by JosephMarinier — APPROVED** [review]
Wow! Good catch! Thank you very much!

**Pull request merged**